src/analyze/analyze.c

```diff
@@ -457,15 +457,13 @@ static int acquire_host_info(sd_bus *bus, struct host_info **hi) {
                                    "/org/freedesktop/hostname1",
                                    hostname_map,
                                    host);
-        if (r < 0)
-                goto fail;
 
-        r = bus_map_all_properties(bus,
+        s = bus_map_all_properties(bus,
                                    "org.freedesktop.systemd1",
                                    "/org/freedesktop/systemd1",
                                    manager_map,
                                    host);
-        if (r < 0)
+        if (r < 0 && s < 0)
```

Comment:
I'd rather explicitly check for the return value of `bus_map_all_properties()` in that case and handle it gracefully.